### ISSUE TITLE
Perf Dashboard: sometimes the new bug flow fails with "No alerts specified to add bugs to."

### ISSUE THREAD

**Issue by tednakamura** [catapult-project/catapult#1830]
Chrome version: 48.0.2564.22 (Windows)
URL: https://chromeperf.appspot.com/alerts?sheriff=WebRTC%20Perf%20Sheriff&sortby=test&sortdirection=down

Please copy and paste any errors from JavaScript console (Ctrl+Shift+J to open):

Please describe the problem:
1. Fill in the checkbox for an alert (or if I'm already viewing a specific graph, click the alerting point)
2. Click the New Bug button. A new window appears to let me file a new bug.
3. (not sure it matters) modify the summary
4. Click the Submit button

Issue:
Bug not filed, and I instead get the error message "No alerts specified to add bugs to."
![screenshot](https://cloud.githubusercontent.com/assets/9001650/11599252/9789d2ba-9a7a-11e5-97ad-e790c70d8935.png)

Reproducibility: sporadic. But today I finally remembered to bring up the debugging pane to try to collect data. If I retry the same steps, a new bug is filed just fine.

Additional data - here's what I see in the Network pane when the bug occurs:
1) POST to https://chromeperf.appspot.com/file_bug returns a 302 to https://accounts.google.com/o/oauth2/auth?state=https%3A%2F%2Fchromeperf.appspot.com%2Ffile_bug%3A_auXP2VO63O8zr1rZGK5tzoxNDQ5MjU3MTcw&redirect_uri=https%3A%2F%2Fchromeperf.appspot.com%2Foauth2callback&response_type=code&client_id=425761728072.apps.googleusercontent.com&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fprojecthosting+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email&access_type=offline

2) GET to https://accounts.google.com/o/oauth2/auth?state=https%3A%2F%2Fchromeperf.appspot.com%2Ffile_bug%3A_auXP2VO63O8zr1rZGK5tzoxNDQ5MjU3MTcw&redirect_uri=https%3A%2F%2Fchromeperf.appspot.com%2Foauth2callback&response_type=code&client_id=425761728072.apps.googleusercontent.com&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fprojecthosting+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email&access_type=offline returns a 302 to https://chromeperf.appspot.com/oauth2callback?state=https://chromeperf.appspot.com/file_bug:_auXP2VO63O8zr1rZGK5tzoxNDQ5MjU3MTcw&code=4/U2g7O99VHkNAIyT2kedI4X3REOScB0lr-yB-F5UzRdM#

3) GET to https://chromeperf.appspot.com/oauth2callback?state=https://chromeperf.appspot.com/file_bug:_auXP2VO63O8zr1rZGK5tzoxNDQ5MjU3MTcw&code=4/U2g7O99VHkNAIyT2kedI4X3REOScB0lr-yB-F5UzRdM returns a 302 to https://chromeperf.appspot.com/file_bug

4) GET to https://chromeperf.appspot.com/file_bug returns a 200 whose response is the html for the above screenshot.

Note that even though I'm 302'd through accounts.google.com, I don't actually receive any UI prompts to reauthenticate. That is, I'm already authenticated, as indicated by the presence of my username in the top right corner. And I don't know what a successful flow looks like because the new window automatically closes, along with the debug pane, when a bug is successfully filed. :)

**Comment by anniesullie** [comment]
@qyearsley @chrisphn

**Comment by qyearsley** [comment]
The flow that is used for filing bugs is a bit interesting: When you click "New Bug", the [triage-dialog](https://github.com/catapult-project/catapult/blob/master/dashboard/dashboard/elements/triage-dialog.html#L377) element opens a new window, and makes a hidden `<form>` element and and then immediately submits that form (to /file_bug), which responds with a new form, with hidden fields. One of the hidden fields is "keys".

Finally, when that form is submitted, if the keys field is empty then we expect the error message "No alerts specified to add bugs to" ([file_bug.py:65](https://github.com/catapult-project/catapult/blob/master/dashboard/dashboard/file_bug.py#L65)).

So, if you ever click "new bug" the hidden field "keys" is empty in the new window (obviously this is hidden but can be viewed using devtools), then the problem must be before that point...

Maybe it would help with diagnosing this in the future if we made it so that the "file new bug" form displayed how many alerts are in the hidden "keys" field?

--------

Of course, this bug filing flow can be improved; one possibility Chris has mentioned is if it didn't open a new window, and the fields (bug title, first comment, labels) were added to the triage-dialog element.
